web/models/audit_log.ex

```diff
@@ -0,0 +1,31 @@
+defmodule HexWeb.AuditLog do
+  use HexWeb.Web, :model
+
+  schema "audit_logs" do
+    belongs_to :actor, User
+    field :action, :string
+    field :extra_params, :map
+    timestamps(updated_at: false)
+  end
+
+  def create(%HexWeb.User{id: user_id}, action, opts) do
+    extra_params = extract_extra_params(action, opts)
+    %HexWeb.AuditLog{actor_id: user_id, action: action, extra_params: extra_params}
+  end
+
+  defp extract_extra_params("docs.publish", release), do: serialize(release)
```

Comment:
@ericmj I just pushed updates. Couple of notes:

- as in previous revisions, I decided to explicitly pattern match on all available actions (docs.publish, docs.revert etc) even though it's not actually needed - but it serves here as kind of live documentation of what we support in just one place
- as you can see below in `fields/1` private function, I'm explicitly whitelisting fields to audit. I thought about using reflection (__schema__/1 etc) to automatically do that, however there are fields that we probably don't want to store like users.password, keys.secret_first etc
- as you can see in controller tests, I somewhat made the assertion more relaxed where I pattern match on a subset of the map (instead of asserting the entire log format, which would make the test somewhat bloated)